styx-common/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java

```diff
@@ -214,44 +214,50 @@ public void authorizeServiceAccountUsage(WorkflowId workflowId, String serviceAc
     @Override
     public ServiceAccountUsageAuthorizationResult checkServiceAccountUsageAuthorization(String serviceAccount,
                                                                                         String principalEmail) {
-      final String projectId = serviceAccountProjectId(serviceAccount);
-
       // First check if principal is blacklisted
       return memberStatus(principalEmail, blacklist)
           .map(status -> ServiceAccountUsageAuthorizationResult.builder()
-              .serviceAccountProjectId(projectId)
               .authorized(false)
               .blacklisted(true)
               .message(blacklistedDenialMessage(principalEmail))
               .build())
           // Then continue authorization check
-          .orElseGet(() -> checkServiceAccountUsageAuthorization0(serviceAccount, principalEmail, projectId));
+          .orElseGet(() -> checkServiceAccountUsageAuthorization0(serviceAccount, principalEmail));
     }
 
     private ServiceAccountUsageAuthorizationResult checkServiceAccountUsageAuthorization0(String serviceAccount,
-                                                                                          String principalEmail,
-                                                                                          String projectId) {
-      final Optional<String> accessMessage = firstPresent(
-          // Check if the principal is an admin
-          () -> memberStatus(principalEmail, administrators)
-              .map(status -> String.format("Principal %s is an admin %s", principalEmail, status)),
-
-          // Check if the principal has been granted the service account user role in the project of the SA
-          () -> projectPolicyAccess(projectId, principalEmail)
-              .map(type -> String.format("Principal %s has role %s in project %s %s",
-                  principalEmail, serviceAccountUserRole, projectId, type)),
-
-          // Check if the principal has been granted the service account user role on the SA itself
-          () -> serviceAccountPolicyAccess(serviceAccount, principalEmail)
-              .map(type -> String.format("Principal %s has role %s on service account %s %s",
-                  principalEmail, serviceAccountUserRole, serviceAccount, type)));
-
-      final ServiceAccountUsageAuthorizationResultBuilder result = ServiceAccountUsageAuthorizationResult.builder()
-              .serviceAccountProjectId(projectId)
-              .authorized(accessMessage.isPresent())
-              .message(accessMessage.orElseGet(() -> denialMessage(serviceAccount, principalEmail, projectId)));
-
-      return result.build();
+                                                                                          String principalEmail) {
```

Comment:
It is weird the division of concerns between `checkServiceAccountUsageAuthorization` and `checkServiceAccountUsageAuthorization0` and the naming also.

What about we keep the public method high level and then we create small private methods for every check. Something like this:
```java
@Override
    public ServiceAccountUsageAuthorizationResult checkServiceAccountUsageAuthorization(String serviceAccount,	    public ServiceAccountUsageAuthorizationResult checkServiceAccountUsageAuthorization(String serviceAccount,
                                                                                        String principalEmail) {	                                                                                        String principalEmail) {
  return checkIsServiceAccountBlacklisted(serviceAccount, principalEmail)
      .or(() -> checkIsPrincipalAdmin(serviceAccount, principalEmail))
      .orElseGet(() -> {
         final String projectId = serviceAccountProjectId(serviceAccount);
         return checkRole(serviceAccount, principalEmail, projectId)
                  .or(() -> denialMessage(serviceAccount, principalEmail, projectId));
       });
}
```

Reply:
We will also be using method with names instead of comments to explain logic